AppDashboard/lib/app_dashboard_data.py

```diff
@@ -647,3 +729,125 @@ def can_upload_apps(self):
     except Exception as err:
       logging.exception(err)
       return False
+
+  def set_dash_layout_settings(self, values=None):
+    """ Saves user settings
+        for customizing the UI of the Dashboard.
+
+    Args:
+      values: A JSON str that defines the layout of the dash page
+        from /ajax/save/layout
+    """
+    user = users.get_current_user()
+    if not user:
+      return
+    email = user.email()
+    try:
+      user_info = self.get_by_id(UserInfo, email)
+      if user_info:
+        try:
+          temp_dict = json.loads(values)
+        except TypeError as err:
+          #base admin template
+          default_value = '''{
+            "nav":["app_management", "appscale_management",
+            "debugging_monitoring"],
+            "panel":["app_console","upload_app","cloud_stats","database_stats",
+            "memcache_stats"]
+            }'''
+          temp_dict = json.loads(default_value)
+        LOOKUP_DICT = self.build_dict(email=email)
+        temp_dict['nav'] = [{key: LOOKUP_DICT.get(key)} for key in
+                            temp_dict.get('nav') if
+                            key in LOOKUP_DICT]
+
+        temp_dict['panel'] = [{key: LOOKUP_DICT.get(key)} for key in
+                              temp_dict.get('panel') if
+                              key in LOOKUP_DICT]
+        user_info.dash_layout_settings = temp_dict
+        user_info.put()
+      return
+    except Exception as err:
+      logging.exception(err)
+      pass
+
+
+  def rebuild_dash_layout_settings_dict(self, email=None):
+    """ Rebuilds the user's layout settings in case there is an update
+      to the lookup dictionary
+
+    Args:
+      email: A str that indicates the e-mail address of the administrator of
+        this application, or None if the currently logged-in user is the admin.
```

Comment:
Okay, the docstring makes is sound like I need to explicitly pass in None in cases where the user is the admin. If I understand correctly, you were trying to indicate that the email parameter is not required if the user is logged in (since it will be auto-populated). I think you can just omit the "or None" clause.